### PR TITLE
Adding connection disconnection success and services resolved callbacks on bluezdbus and dotnet

### PR DESCRIPTION
**Pull request by kent-williams**
- [x] bluezdbus

- [x] dotnet

### CODE REVIEW

File: aioble/bluezdbus/device.py [hunk 1]
```diff
@@ -55,16 +67,41 @@ def __init__(self, address, loop=None):
...
-        """Interface Added Signal"""
+    async def _disconnect_succeeded(self):
+        # Call User Callback
+        self.disconnect_succeeded()
+
+        # THROWING SIGSEGV RIGHT NOW
+        # Add Match Callback for PropertiesChanged
+        #await self._dbus.bus_remove_match_action_async(self.properties_rule, self.properties_changed, None)
+
+    def _services_resolved(self):
```
Comment (jtdreisb) [inline aioble/bluezdbus/device.py]: it doesn't look like this is being called.

File: aioble/bluezdbus/device.py [hunk 2]
```diff
@@ -267,6 +305,10 @@ def signal_parser(self, connection, message, data):
             for service in self.services:
                 await service.resolve_characteristics()
 
+        # Notify User that Services have been Discovered
+        self.services_resolved()
```
Comment (jtdreisb) [inline aioble/bluezdbus/device.py]: We should probably be calling self._services_resolved() and put logic inside the helper method that checks for NoneType before invoking the callback.

File: aioble/bluezdbus/device.py [hunk 3]
```diff
@@ -267,6 +305,10 @@ def signal_parser(self, connection, message, data):
...
+        self.is_services_resolved = True
```
Comment (jtdreisb) [inline aioble/bluezdbus/device.py]: should is_services_resolved be set to true before we call our callback?

**Comment by kent-williams** [comment]
Couldn't get connect/disconnect events working on dotnet, skipping for now.

**Pull request merged**